src/main/java/io/vertx/openapi/validation/impl/RequestValidatorImpl.java

```diff
@@ -153,6 +154,28 @@ RequestParameter validateBody(RequestBody requestBody, ValidatableRequest reques
       throw new ValidatorException("The format of the request body is not supported", UNSUPPORTED_VALUE_FORMAT);
     }
     Object transformedValue = transformer.transformRequest(mediaType, request);
+    // Corner case for "application/octet-stream" media type:
+    // Skip full schema validation if the following simplified validation succeeds,
+    // otherwise validate with JSON Validator and fail the validation
+    // of the transformed value eventually.
+    if (mediaType.getIdentifier().equals(MediaType.APPLICATION_OCTET_STREAM)) {
+      JsonSchema schema = mediaType.getSchema();
+      String schemaTypeValue = schema.get("type", null);
```

Comment:
You will always get a `type`, because in case that schema is empty or null, we replace it with a default

Reply:
I simplify that one.